common/src/test/java/com/commercetools/payment/methods/CreatePaymentTransactionMethodBaseTest.java

```diff
@@ -0,0 +1,122 @@
+
+    @Mock
+    private SphereClient sphereClient;
+
+    @Mock
+    private Payment payment;
+
+    private CreatePaymentTransactionData transactionData;
+
+    private MonetaryAmount amount42;
+
+    @Before
+    public void setUp() {
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put("key1", "value1");
+        configMap.put("key2", "value2");
+
+        amount42 = Money.of(42.42, "EUR");
+
+        when(payment.getAmountPlanned()).thenReturn(amount42);
+
+        transactionData = new CreatePaymentTransactionDataImpl(sphereClient, "mock-ref", configMap);
+        transactionData.setPayment(payment);
+
+        doCallRealMethod().when(createPaymentTransactionMethod).createPaymentTransaction(any(), any());
+        doCallRealMethod().when(createPaymentTransactionMethod).createTransactionDraftBuilder(any(), any());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void createPaymentTransaction_sendsPaymentUpdateCommandWithInitialTransactionState() {
+        createPaymentTransactionMethod.createPaymentTransaction(transactionData, AUTHORIZATION);
+
+        ArgumentCaptor<PaymentUpdateCommand> updateCommand = ArgumentCaptor.forClass(PaymentUpdateCommand.class);
+        verify(sphereClient, times(1)).execute(updateCommand.capture());
+
+        assertThat(updateCommand.getValue().getUpdateActions())
+                .filteredOn(action -> action instanceof AddTransaction)
+                .extracting(action -> ((AddTransaction) action).getTransaction().getState())
+                .containsExactly(INITIAL);
+    }
+
+    @Test
+    public void createTransactionDraftBuilder_simpleCase() {
+        TransactionDraft transactionDraft = createPaymentTransactionMethod.createTransactionDraftBuilder(transactionData, AUTHORIZATION).build();
+        assertThat(transactionDraft).isNotNull();
+        assertThat(transactionDraft.getTimestamp()).isNull();
+        assertThat(transactionDraft.getInteractionId()).isNull();
+        assertThat(transactionDraft.getAmount()).isEqualByComparingTo(Money.of(42.42, "EUR"));
+        assertThat(transactionDraft.getType()).isEqualByComparingTo(AUTHORIZATION);
+        assertThat(transactionDraft.getState()).isEqualByComparingTo(INITIAL);
+
+        transactionDraft = createPaymentTransactionMethod.createTransactionDraftBuilder(transactionData, CHARGE).build();
+        assertThat(transactionDraft).isNotNull();
+        assertThat(transactionDraft.getTimestamp()).isNull();
+        assertThat(transactionDraft.getInteractionId()).isNull();
+        assertThat(transactionDraft.getAmount()).isEqualByComparingTo(Money.of(42.42, "EUR"));
+        assertThat(transactionDraft.getType()).isEqualByComparingTo(CHARGE);
+        assertThat(transactionDraft.getState()).isEqualByComparingTo(INITIAL);
+    }
+
+    @Test
+    public void whenTransactionTypeIsSetInData_createTransactionDraftBuilder_usesValueFromData() {
+        transactionData.setTransactionType(CHARGE);
```

Comment:
I almost oversaw this line and was trying to find the difference in this test and the one below. So maybe an empty line below will make it more clear to see :)

Reply:
indeed, thx:
0a59cab